component/otelcol/internal/scheduler/scheduler.go

```diff
@@ -0,0 +1,150 @@
+// Package scheduler exposes utilities for scheduling and running OpenTelemetry
+// Collector components.
+package scheduler
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/agent/component"
+	otelcomponent "go.opentelemetry.io/collector/component"
+	"go.uber.org/multierr"
+)
+
+// Scheduler implements manages a set of OpenTelemetry Collector components.
+// Scheduler is intended to be used from Flow components which need to schedule
+// OpenTelemetry Collector components; it does not implement the full
+// component.Component interface.
+//
+// Each OpenTelemetry Collector component has one instance per supported
+// telemetry signal, hence supporting multiple OpenTelemetry Collector
+// components inside the scheduler. Scheduler should only be used to manage
+// multiple instances of the same OpenTelemetry Collector component.
```

Comment:
@rfratto I'll read on and maybe things get clearer, but I'm trying to understand what we mean here.

An OpenTelemetry Collector components can support all three of metrics/logs/traces at the same time, right? What do we mean by the following phrase? Should we use one scheduler for multiple OTel Collector component for logs, and then for example, _another_ scheduler for the rest of OTel Collector component for traces+metrics?

> Scheduler should only be used to manage multiple instances of the _same_ OpenTelemetry Collector component.

Reply:
Ah, sorry, I'll try to make this clearer over in #2227. 

It probably should say something like: 

> Schedule should only be use to manage the various telemetry signals of the same OpenTelemetry Collector Component.

Would that make more sense? (for example, use the schedule to manage the logs instance, metrics instance, and traces instance for `otelcol.receiver.otlp`)

Reply:
Is "otlp" here the label, right? So if we have an `otelcol.receiver.prod` and `otelcol.receiver.dev`, we'd have to have one scheduler for each?

Yes then, it makes sense in that context.

Reply:
`otelcol.receiver.otlp` is the component name :) But yes, that still applies. I'll make some tweaks in #2227 and ping you there.